packages/devtools-reps/src/reps/document-type.js

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// ReactJS
+const PropTypes = require("prop-types");
+
+// Reps
+const {
+  getGripType,
+  isGrip,
+  wrapRender,
+} = require("./rep-utils");
+const dom = require("react-dom-factories");
+const { span } = dom;
+
+/**
+ * Renders DOM documentType object.
+ */
+DocumentType.propTypes = {
+  object: PropTypes.object.isRequired,
+};
+
+function DocumentType(props) {
+  const { object } = props;
+  let name = object && object.preview && object.preview.nodeName
+    ? ` ${object.preview.nodeName}`
+    : "";
+  return span({
+    "data-link-actor-id": props.object.actor,
+    className: "objectBox objectBox-document"
+  }, `<!DOCTYPE${name}>`);
+}
+
+// Registration
+function supportsObject(object, noGrip = false) {
+  if (noGrip === true || !isGrip(object)) {
+    return false;
+  }
+
+  const type = getGripType(object, noGrip);
+  return (object.preview && type === "DocumentType");
+}
+
+// Exports from this module
+module.exports = {
+  rep: wrapRender(DocumentType),
```

Comment:
Just a note that `wrapRender` util might be replaced by using Error Boundaries introduced in React16
https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html
Of course, out of scope of this bug.

Reply:
Yes, that would be nice.
Also, this would mean having a stateful component (e.g. top level Rep) to have the `componentDidThrow` lifecycle method. We'll have to check if it does not harm the performances too much.